sky/jobs/utils.py

```diff
@@ -448,18 +449,48 @@ def stream_logs_by_id(job_id: int, follow: bool = True) -> str:
     return ''
 
 
-def stream_logs_by_name(job_name: str, follow: bool = True) -> str:
-    """Stream logs by name."""
-    job_ids = managed_job_state.get_nonterminal_job_ids_by_name(job_name)
-    if len(job_ids) == 0:
-        return (f'{colorama.Fore.RED}No job found with name {job_name!r}.'
-                f'{colorama.Style.RESET_ALL}')
-    if len(job_ids) > 1:
-        return (f'{colorama.Fore.RED}Multiple running jobs found '
-                f'with name {job_name!r}.\n'
-                f'Job IDs: {job_ids}{colorama.Style.RESET_ALL}')
-    stream_logs_by_id(job_ids[0], follow)
-    return ''
+def stream_logs(job_id: Optional[int],
+                job_name: Optional[str],
+                controller: bool = False,
+                follow: bool = True) -> str:
+    """Stream logs by job id or job name."""
+    if job_id is None and job_name is None:
+        job_id = managed_job_state.get_latest_job_id()
+        if job_id is None:
+            return 'No managed job found.'
+    if controller:
+        if job_id is None:
+            assert job_name is not None
+            managed_jobs = managed_job_state.get_managed_jobs()
+            managed_jobs = list(
+                filter(lambda job: job['job_name'] == job_name, managed_jobs))
```

Comment:
I'm wondering if this could be replaced by a database operation for optimization. Maybe left an TODO here?

Reply:
Actually, seems like the `get_nonterminal_job_ids_by_name` already implemented this?

Reply:
No, `get_nonterminal_job_ids_by_name` only returns the jobs with **nonterminal state**, but we are fetching all the jobs that has the name as `--controller` should be callable for terminated job. Added a comment here.

Reply:
Good point! Agree. LGTM!